### PR TITLE
Adding test cases for About component

### LINKED ISSUE #33: Add test for About.tsx

### PR DESCRIPTION
**Pull request by MohitPopli**
Screen Shot 2021-02-27 at 2 49 24 PM

Closes #33

### CODE REVIEW

File: src/components/about/About.test.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    <Router history={history}>
+      <About />
+    </Router>
+  );
+
+  const navButton = container.querySelector(
+    ".navbar__back-button"
+  ) as HTMLAnchorElement;
+
+  fireEvent.click(navButton);
```
Comment (emma-r-slight) [inline src/components/about/About.test.tsx]: you could use the user Event library here instead of fireEvent...

https://testing-library.com/docs/ecosystem-user-event/

Comment (MohitPopli) [reply src/components/about/About.test.tsx]: I see. I thought here fireEvent would be okay. But I replaced it with userEvent to be consistent
Thanks.

File: src/components/about/About.test.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  );
+  expect(container.querySelector(".about__subtitle")?.textContent).toEqual(
+    "App Settings"
+  );
+  expect(iconButtons[0]?.textContent).toEqual("Change language");
+  expect(iconButtons[1]?.textContent).toEqual("Change Theme");
+  expect(iconButtons[2]?.textContent).toEqual("Contact me");
+  expect(iconButtons[3]?.textContent).toEqual("Source code");
+  expect(iconButtons[4]?.textContent).toEqual("Report bug");
+});
```
Comment (emma-r-slight) [inline src/components/about/About.test.tsx]: if you add `  screen.logTestingPlaygroundURL()` to the test after you render the component.  You will be able to find all the elements that can be selected and tested against.

this is the result for the about page

https://testing-playground.com/#markup=DwEwlgbgfKkAQGMA2BDAzmgvAIhQIwHsBXAF2xnAkVQxwDsUI8UAnOfYkgfS4adfLAU1dFmwBZImABCpEgTrT0AUwC0LAgRJw8chXDAIFq3SXl04fZix7MEAaxN662OCXxg6IZQA8cABlcNJGUcU3NXVjAUVXA0fBCQHAAzFCQ0ZVcQFHdVHIIAWyczBRwSFiJMinhkURxDY3CFHgaXGDQABxQLWtpsVwB3MBASAAscACYAFldR5TAAc1GSSZmYAHpO7o3Kdq6emjFJMAAVYgRRgCUwDo6Q9U0yDa26DZRqql6xK1YeEjASCFyABBQikYDrXYQqGUER9BCsEDsMHcLgoW6qDosZTJZTYugIZRoQSwr44DikHhoIh4f6AqrA25wADKyjMngWaGhkA+cLEFNR6I6mOxuPxhLQtmcxJgwjJEikshKihUDy0OmcBiMdGK5jgSAICPuGRCCHkNi4TRcbg8Xl8ASCBBCYWckRY0ViYHieESKTSGSyORi+SKVrKFSqsE+h3q2t1zS4rUELz5OEGwzGq1m8yWK2w00Em32Ox5UdT2Fa8bof18TwAwqNugtlPqm0QUM3udBgCn5cczkQLtdbvcNFpCy83rLy8cleYlBk1dorVrGq6bXhPN4-NhAnBgqFsFa3R64gllElsKl0pk4NlciGq+HKiSajGK3GrS1tcn9uX0yM4z5jMcBzIsyxZs8xZdry8qVl+XAkLW5ANk2LYnHMBTKDBPZ-n2UgDkONx3GoY5PBCk4Qu8OGQqWpLvgiLBIgKPACq+0Z1LgKKsdxlrSoIcrvrOzgLqRjwasqq46seG5bvau4njEZ4+hefo3oGD7yKGzjPreoyijgywkB0aAAFzrOsJBDGYeIAHRGAU6wABIEBUaAAOLEEg7HlvBzjfgov7dP+cBDIBWagTmEHARO0G0d29GcX5yo1j49YKO4ZpwFhOG9kJBHnFcxGjo8sXbFR074TIImqmREl6slESyXaO57lESleuel7XgGd5BnkWlPtg5QvqBBnYEZJnmesCwAqMNL2YUzmuUQHleesj4dAMKA+XBn7+YmP57MF8oAZmMWReBeYFlB5XxbB76NQmSFpeQzLECwhKIAQ3i5Xh+WnIVw4kUuZWvBVQgzoqNWLnVK5Pda7ibi1DrsO6HXer6V7+re97BoNYbDRGszjQUKBgEg8imaMK1oAsXkAAILGTFOLQUu2PftKWHYFx0HJxZ1AQWl25pBFFxTCb5JVz5ipU8lzKB0rnLkQCx-SdAOEUVI5ieOt3gxtJbdvdNG7EAA

..click on the element and you can copy the selector to test against.  

eg for example it shows you can use this  `screen.getByText(/change language/i)`  instead of a queryselector to get the icon text

Comment (MohitPopli) [reply src/components/about/About.test.tsx]: Such a wonderful suggestion. Thanks

File: src/components/about/About.test.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    "href",
+    "mailto:horusgoul@gmail.com"
+  );
+});
+
+test("should invoke onNavbarBackButtonClick", () => {
+  const history = createMemoryHistory({
+    initialEntries: ["/about"],
+  });
+  history.action = "PUSH";
```
Comment (mpeyper) [inline src/components/about/About.test.tsx]: Why is this required?

File: src/components/about/About.test.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+test("should render component", () => {
+  const history = createMemoryHistory({
+    initialEntries: ["/about"],
+  });
+  render(
+    <Router history={history}>
+      <About />
+    </Router>
+  );
```
Comment (mpeyper) [inline src/components/about/About.test.tsx]: Minor thing, but I found the `Router` and `history` weren't actually required unless you were clicking the links so you might be able to simplify this test (and possibly others) a bit if you wanted to with:

```suggestion
  render(<About />);
```

File: src/components/about/About.test.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+  expect(screen.getByText(/about/i)).toBeInTheDocument();
+  expect(screen.getByText(/app settings/i)).toBeInTheDocument();
+  expect(screen.getByText(/change language/i)).toBeInTheDocument();
+  expect(screen.getByText(/change theme/i)).toBeInTheDocument();
+  expect(screen.getByText(/contact me/i)).toBeInTheDocument();
+  expect(screen.getByText(/source code/i)).toBeInTheDocument();
+  expect(screen.getByText(/report bug/i)).toBeInTheDocument();
```
Comment (mpeyper) [inline src/components/about/About.test.tsx]: you could consider using [`jest-in-case` and making seperate cases](https://github.com/OpenSourceRaidGuild/atom-pwa/blob/next/src/components/main-menu/MainMenu.test.tsx#L21-L60) for each of these rather than a single test for them all

**Review by mpeyper — APPROVED** [review]
Some minor suggestions made, but happy to accept them if you don't want to or disagree with any of them.

File: src/components/about/About.test.tsx (hunk 4 above)
Comment (MohitPopli) [reply src/components/about/About.test.tsx]: I doubt as I tried removing router, but the component itself uses `withRouter` HOC which will force me to wrap my component in jest otherwise my test will fail,

`Invariant failed: You should not use <withRouter(About) /> outside a <Router>`
This is the error that I'm getting if I don't wrap it with router. This might not be required once we start the refactoring phase but I'm not 100% sure right now

File: src/components/about/About.test.tsx (hunk 3 above)
Comment (MohitPopli) [reply src/components/about/About.test.tsx]: Ya I'll remove this. Actually was trying some other approach and it was left. It's not needed anymore. Thanks for spotting it.

File: src/components/about/About.test.tsx (hunk 5 above)
Comment (MohitPopli) [reply src/components/about/About.test.tsx]: Thanks for this nice suggestion. I have implemented the same.
